grpc/src/main/java/com/linecorp/armeria/server/grpc/DeferredListener.java

```diff
@@ -75,16 +71,16 @@
 
             this.delegate = delegate;
             try {
-                for (Consumer<Listener<I>> task : pendingTasks) {
-                    task.accept(delegate);
+                while (!pendingTasks.isEmpty()) {
+                    pendingTasks.poll().accept(delegate);
                 }
```

Comment:
I'm not 100 sure my approach is better. But we can allocate a new empty array before iterating `pendingTasks`.
```
for (;;) {
final List<Consumer<Listener<I>>> pendingTasks = this.pendingTasks;
if (pendingTasks.isEmpty()) {
    break;
}

// New pending tasks could be added while invoking pending tasks.
this.pendingTasks = new ArrayList<>();
try {
    for (Consumer<Listener<I>> task : pendingTasks) {
        task.accept(delegate);
    }
} catch (Throwable ex) {
    callClosed = true;
    armeriaServerCall.close(ex);
    return null;
}
```

Reply:
Don't we need to consider thread-safe type?